test/test_compress.cpp

```diff
@@ -34,7 +34,8 @@ TEST_CASE("Compress index", "[index][compress]") {
         (tmp.path() / encoding).string(),
         ScorerParams(""),  // no scorer
         std::nullopt,  // no quantization
-        true  // check=true
+        true,  // check=true
+        false  // in-memory=false
```

Comment:
Let's cover both cases here. We can do this with:

```cpp
bool in_memory = GENERATE(true, false);
```

similar to `encoding` above, and then passing that value to `compress` function.